src/olympia/promoted/tasks.py

```diff
@@ -106,3 +109,65 @@ def on_stripe_charge_failed(event):
         from_email=settings.ADDONS_EMAIL,
         recipient_list=["amo-admins@mozilla.com"],
     )
+
+
+@task
+def on_stripe_customer_subscription_deleted(event):
+    event_id = event.get("id")
+    event_type = event.get("type")
+
+    if event_type != "customer.subscription.deleted":
+        log.error(
+            'invalid event "%s" received (event_id=%s).', event_type, event_id
+        )
+        return
+
+    # This event should contain a `subscription` object.
+    subscription = event.get("data", {}).get("object")
+    if not subscription:
+        log.error("no subscription object in event (event_id=%s).", event_id)
+        return
+
+    subscription_id = subscription["id"]
+    log.info(
+        'received "%s" event with subscription_id=%s.',
+        event_type,
+        subscription_id,
+    )
+
+    try:
+        sub = PromotedSubscription.objects.get(
+            stripe_subscription_id=subscription_id
+        )
+    except PromotedSubscription.DoesNotExist:
+        log.error(
+            'received a "%s" event (event_id=%s) for a non-existent'
+            " promoted subscription (subscription_id=%s).",
```

Comment:
nits: space at the end of lines rather than start; mixed `"` and `'` for no apparent reason.

Reply:
Quotes are added/adjusted by Black, which I run locally on new files I create.

Reply:
I am still hoping to adopt Black soon-ish so I let you folks prepare for that given that Black isn't going to support single quotes (it enforces double quotes most of the time)

Reply:
I half read through that issue before getting bored - I summarise it as "we made a decision and won't back down out of stubbornness".  It's clear that single quotes are the python default:
```python
>>> foo = "foo"
>>> foo
'foo'
```
😛 (in addition to the many other examples in the issue)

I strongly suspect if we do ever deploy Black on this repo, it'll be with `--skip-string-normalization`.

Reply:
> I summarise it as "we made a decision and won't back down out of stubbornness". 

I agree.

> I strongly suspect if we do ever deploy Black on this repo, it'll be with --skip-string-normalization.

I'd prefer not to skip that, otherwise why having a code formatter if we don't use it fully? Anyway, we'll see if I can make another random act of kindness this quarter :)